src/components/intent.ts

```diff
@@ -575,6 +575,17 @@ export class Intent {
         return this.client.setAvatarUrl(url);
     }
 
+    public async setRoomUserDisplayName(roomId: string, name: string) {
```

Comment:
I'd make this generic and allow people to set their avatar too (possibly with a key: `key: "displayname"|"avatar_url", value: string`)?